### PR TITLE
prov/sockets: Use `ofi_total_ioc_cnt` instead of self-implemented `for` statement

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
- Some cosmetic code cleanups to reduce code amount (fi_atomic only)
- self-implemented `for` - > `ofi_total_ioc_cnt` function: Cosmetic changes to align the implementation of the `sock_ep_atomic_writev` with the `sock_ep_atomic_readwritev` and `sock_ep_atomic_compwritev` functions

Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

**Comment by shefty** [comment]
Please try to limit changes to only what's needed.  There's a lot of changing of white space that isn't necessarily an improvement.  Also, limit use of inline.  The compiler is free to ignore it anyway, and increasing code size can be worse than condensed code with function calls because of the impact on CPU caches.

File: prov/sockets/src/sock_atomic.c [hunk 1]
```diff
@@ -391,86 +374,73 @@ static ssize_t sock_ep_atomic_readwritemsg(struct fid_ep *ep,
 		return -FI_EINVAL;
 	}
 
-	return sock_ep_tx_atomic(ep, msg, NULL, NULL, 0,
-				 resultv, result_desc, result_count, flags);
+	return sock_ep_tx_atomic(ep, msg,
+				 NULL, NULL, 0,
+				 resultv, result_desc, result_count,
+				 flags);
```
Comment (shefty) [inline prov/sockets/src/sock_atomic.c]: unnecessary change

**Comment by gladkovdmitry17** [comment]
> Please try to limit changes to only what's needed.

Yes, will do that by adding changes to the separated commits.



> Also, limit use of inline. The compiler is free to ignore it anyway, and increasing code size can be worse than condensed code with function calls because of the impact on CPU caches.

Sure, the compiler can ignore it. But from my understanding, I just replaced a part of code by inlined function (if the compiler is going to inline this function). Seems amount of code will be almost equal, since the functions (impacted by my changes) is called directly only in one place.

Thanks for commenting this out. I'll take it into account for further PRs.

**Comment by shefty** [comment]
The calls that you marked as inline are assigned to function pointers, which are used by the external headers.  So they are required to exist as stand-alone functions.  The inline is adding a new copy of the code into the other locations, which are also assigned function pointers.  If the compiler fully inlines the code, there would be 3 copies of the *msg() function as a result.  That's my concern.

**Comment by gladkovdmitry17** [comment]
> The calls that you marked as inline are assigned to function pointers, which are used by the external headers.

Thanks for the explanation. It's right concern. I had no thoughts regarding this point. Should I return back the previous changes?

**Comment by gladkovdmitry17** [comment]
> Should I return back the previous changes?

@shefty I've returned back the calling of *msg() function instead of *v().
